### ISSUE TITLE
[Bug]: Prod. Order Components - Item Substitution action issue

### ISSUE THREAD

**Issue by GregP-Navdna** [microsoft/ALAppExtensions#25298]
### Describe the issue

the following page appears to be causing an issue. when using the Item Substituion action. a client came across this in version 22.5. I looked at version 23 and it is still there. 

page 99000818 "Prod. Order Components"
                action(SelectItemSubstitution)
                {
                    AccessByPermission = TableData "Item Substitution" = R;
                    ApplicationArea = Manufacturing;
                    Caption = '&Select Item Substitution';
                    Image = SelectItemSubstitution;
                    ToolTip = 'Select another item that has been set up to be traded instead of the original item if it is unavailable.';

                    trigger OnAction()
                    begin
                        CurrPage.SaveRecord();
                        ShowItemSub();
                        CurrPage.Update(true);
                        ReserveComp();
                    end;
                }
                
            

   the following error occurs when attempting to trigger the action. 

An error occurred and the transaction is stopped. Contact your administrator or partner for further assistance.

The following AL methods are limited during write transactions because one or more tables will be locked:

Form.RunModal is not allowed in write transactions.

Codeunit.Run is allowed in write transactions only if the return value is not used. For example, 'OK := Codeunit.Run()' is not allowed.

Report.RunModal is allowed in write transactions only if 'RequestForm = false'. For example, 'Report.RunModal(...,false)' is allowed.

XmlPort.RunModal is allowed in write transactions only if 'RequestForm = false'. For example, 'XmlPort.RunModal(...,false)' is allowed.

Use the commit method to save the changes before this call, or structure the code differently.

Contact your application developer for further assistance.
AL call stack:

"Item Subst."(CodeUnit 5701).GetCompSubst line 19 - Base Application by Microsoft

"Prod. Order Component"(Table 5407).ShowItemSub line 9 - Base Application by Microsoft

"Prod. Order Components"(Page 99000818)."SelectItemSubstitution - OnAction"(Trigger) line 3 - Base Application by Microsoft

### Expected behavior

the item substitution page is supposed to open

### Steps to reproduce

click the action on the page

### Additional context

I was able to make a workaround for them by removing the currpageupdate and the saverecord. since the page is saving the values already this is probably just residual code form NAV that wasn't updated. if it wasn't and I'm completely wrong then let me know but the error occurs when the lines are there, once removed there is no error and the action works as expected.

**Comment by JesperSchulz** [comment]
Great catch! And thank you for reporting the issue. Unfortunately, this repository isn't the right place to report your bug.

In this repository you can create issues concerning the extensibility of Business Central, such as missing events. You can also create issues you would like to fix yourself in the code present in this repository, which are our 1st party apps.

Your bug seems to be BaseApp code though, which only can get addressed in our [BaseApp contribution pilot](https://github.com/microsoft/BusinessCentralApps). Do you want me to transfer this issue to the BaseApp pilot repository?

Note, that the transferred issue would have to get approved in triage and you would have to create a pull request yourself. GitHub is not the right place to report issues you would like Microsoft to address. For that, please create a support request.

PS: You need to be granted access to the pilot repo, if you would like to collaborate on the BaseApp with us. To get access, please fill out the following form: https://forms.office.com/r/E9wyE5a58f.

PPS: I recognize that the process on where you can report which issues is a bit messy these days. We're in between processes and hope to be able to streamline and document the right processes in the very near future.

**Comment by GregP-Navdna** [comment]
yes please transfer if you could. I've filled out the form stated. Thanks for the follow up and explanation.

**Comment by JesperSchulz** [comment]
You should now have access to the new repository! I will transfer this issue right away. We will soon triage it. Once approved, you are ready to create your first pull request here on GitHub for the Base Application 🥳
